### DISCUSSION TITLE
Monitor calibration

### DISCUSSION THREAD

**Discussion by mecui22** [jspsych/jsPsych#1485] [General]
Hi - this question might have been brought up a hundred times...

Is there a plugin for monitor calibration?
For example, I am currently using the credit card test and/or the jspsych resize plugin to control the size of my stim; and using the blindspot test as a virtual chinrest. 
Additionally, I hope to record each participant's monitor refreshing rate, brightness, contrast and dimensions. It would even be a better idea to have some calibration test so participants can manually change the brightness and contrast of the monitor to a certain range. Any suggestions on that? Or this is too ambitious for an online experiment now?

Cheers!

**Comment by jodeleeuw** [comment]
Hi @mecui22,

I'm not sure if this is too ambitious. Some of these are doable, others I'm not sure about.

* You can get an estimate of refresh rate by checking how frequently `requestAnimationFrame()` is called. If you search for JavaScript FPS estimation I'm sure you'll find code. 
* Dimensions of the browser window in pixels are certainly obtainable, and if you use the resize plugin then you'll also know the pixels-to-physical-units conversion rate. 
* I know that you can't record brightness and contrast directly. JavaScript doesn't provide that level of hardware information. I don't know if some kind of participant-led calibration procedure could be used here. I don't really know how people do this in the lab, so I don't know what's realistic for online. If you come up with something and want to turn it into a plugin let us know :grin:.
